sqlmesh/core/table_diff.py

```diff
@@ -313,14 +328,16 @@ def name(e: exp.Expression) -> str:
                     ).as_("null_grain"),
                     *comparisons,
                 )
-                .from_(exp.alias_(self.source, "s"))
-                .join(
-                    self.target,
-                    on=self.on,
-                    join_type="FULL",
-                    join_alias="t",
-                )
-                .where(self.where)
+                .from_(exp.alias_(source_identifier, "s"))
+                .join(exp.alias_(target_identifier, "t"), on=self.on, join_type="FULL")
+            )
+
```

Comment:
i don’t like manually creating with, can you use a builder api or use subqueries?

Reply:
Is the builder API documented anywhere? There didnt seem to be a `.with(..)`

Reply:
ah, I think I found it. There appears to be no top-level `with_()` like there is for `select()` but doing `exp.Select().with_(...)` gets me there.

Is that better?

Reply:
yea